tests/app/main/test_suppliers.py

```diff
@@ -182,24 +169,30 @@ def test_shows_edit_buttons(self, get_current_suppliers_users, data_api_client):
             document = html.fromstring(res.get_data(as_text=True))
 
             assert document.xpath(
-                "//a[normalize-space(string())=$t][@href=$u][contains(@class, $c)]",
-                t="Edit",
-                u="/suppliers/edit",
-                c="summary-change-link",
-            )
-
-            assert document.xpath(
-                "//tr[./td[normalize-space(string())=$f]][.//a[normalize-space(string())=$t][@href=$u]]",
+                "//*[(.//h3)[1][normalize-space(string())=$f]][.//a[normalize-space(string())=$t][@href=$u]]",
                 f="G-Cloud 6",
                 t="View services",
                 u="/suppliers/frameworks/g-cloud-6/services",
             )
+
             assert not document.xpath(
-                "//tr[./td[normalize-space(string())=$f]][.//a[normalize-space(string())=$t]]",
+                "//*[(.//h3)[1][normalize-space(string())=$f]][.//a[normalize-space(string())=$t]]",
                 f="G-Cloud 7",
                 t="View services",
             )
             assert not document.xpath("//a[@href=$u]", u="/suppliers/frameworks/g-cloud-7/services")
+            assert not document.xpath(
```

Comment:
Arrrgh actually this one turns out to be really hard to slacken and still have the right meaning. It looks like it's hyper-specific, but it's actually just saying "there isn't an element with an `h3` title of 'H-Cloud 88' which also contains a 'View services' link".